pkg/sentry/arch/arch_arm64.s

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "textflag.h"
+
+#define FPCR_RM_RN (0x0 << 22)	// Round to Nearest(RN) mode
+#define FPCR_INIT FPCR_RM_RN
+
+// initAarch64FPState initializes floating point state.
+//
+// func initAarch64FPState(data *FloatingPointData)
+//
+TEXT ·initAarch64FPState(SB),NOSPLIT,$0-8
```

Comment:
Does this function initialize the fpu state of the current process? The function argument isn't used, is it?

Reply:
The data argument here points on a floating point state save area  and it is where the state has to be initialized.

Reply:
Hi @avagin 
This floating point state can be initialized to all 0.
And we don't need to initialize the memory.

FPCR(FP control register) and all Vregs are initialized to 0 for gVisor.
FPCR = FPCR_RM_RN (0x0 << 22).

Regarding to FP head(Magic/Size), Magic/Size are only used in Linux kernel for now.
We don't need to set them in gVisor.
Becasue the ptrace API(setFPRegs/getFPRegs) does not need the FP head.
And in kvm platform, about the test case 'TestKernelFloatingPoint', the FP state was captured from sigaction(bluepill). And for other container application test cases in KVM, we only need to save/restore FPSR/FPCR/Vregs.

I have tested the related FP cases with ptrace/kvm platform.

Reply:
It is good that a floating point state can be initialized to all 0, but why do you need to change the sentry floating point state in this function?

Reply:
Hi @avagin 
My idea is to refer to the practices of Linux kernel x86/fpu__clear() and arm64/fpsimd_flush_thread.
There are two steps in fpu__clear() :
1, init the fpstate in memory
2, init fpregs： copy_init_fpstate_to_fpregs();

Reply:
hmmm...
Let me remove the step2....

Reply:
Hi @avagin 
Done.
By the way, kvm and ptrace platforms have different structure definitions for Fpstate on Arm64. But their length is the same.
ptrace:  user_fpsimd_state in linux/arch/arm64/include/uapi/asm/ptrace.h
struct user_fpsimd_state {
        __uint128_t     vregs[32];
        __u32           fpsr;
        __u32           fpcr;
        __u32           __reserved[2];
};

KVM: fpsimd_context in  linux/arch/arm64/include/uapi/asm/sigcontext.h
struct fpsimd_context {
        struct _aarch64_ctx head;
        __u32 fpsr;
        __u32 fpcr;
        __uint128_t vregs[32];
};

So, if fp head is useless in sentry/kvm/ptrace, we should not set them.